tests/suite/test_virtual_server_grpc.py

```diff
@@ -175,10 +175,11 @@ def test_grpc_error_intercept(self, kube_apis, ingress_controller_prerequisites,
                 print(e)
         # Assert the grpc_status is also in the logs.
         ic_pod_name = get_first_pod_name(kube_apis.v1, ingress_controller_prerequisites.namespace)
-        wait_before_test(1)
-        last_log = get_last_log_entry(kube_apis.v1, ic_pod_name, ingress_controller_prerequisites.namespace)
-        assert '"POST /helloworld.Greeter/SayHello HTTP/2.0" 204 14' in last_log
-    
+        wait_before_test()
+        # Need to get full log because of a race condition on the last log entry.
+        logs = kube_apis.v1.read_namespaced_pod_log(ic_pod_name, ingress_controller_prerequisites.namespace)
```

Comment:
still taking long enough to not get logged it seems

Reply:
something like this might help:
```
retry = 0
while '"POST /helloworld.Greeter/SayHello HTTP/2.0" 204 14' not in log_contents and retry <= 60:
            log_contents_block = get_file_contents(
                kube_apis.v1, log_loc, syslog_pod, test_namespace
            )
            retry += 1
            wait_before_test(1)
            print(f"Security log not updated, retrying... #{retry}")
```

Reply:
Thanks, that seems to be working now!